### ISSUE TITLE
Allow customization of conda channels in use

### ISSUE THREAD

**Issue by jaimergp** [napari/napari-plugin-manager#78]
One request I would like to add.

- [ ] The developer defines the channels (other than conda-forge ) proposed for the plugin, and the GUI allows the user to activate those channels with a single click when they install it.

This is very useful for plugins that rely on very important but released libraries in their own channel, such as pytorch, for example.

_Originally posted by @hiroalchem in https://github.com/napari/napari-plugin-manager/issues/14#issuecomment-1611532586_

**Comment by jaimergp** [comment]
> @hiroalchem question, is your request gear towards the lineedit in the bottom of the dialog where you can type any package or drag and drop a tarball from the filesystem?
> 
> Do you mean to add extra channel besides conda-forge, is the case for plugins in the napari plugin ecosystem that they need extra channels besides conda forge?
> 
> Thanks!

Originally posted by @goanpeca in https://github.com/napari/napari-plugin-manager/issues/14#issuecomment-2219545407

**Comment by goanpeca** [comment]
Haven't heard from @hiroalchem, so I am not sure I follow. Is this about adding extra channels that will be added globally to any conda install, or locally (per package item) when installing something?

Thoughts @jaimergp ?

**Comment by jaimergp** [comment]
I think they'd like to obtain plugins for non-conda-forge channels? We currently feed it from the npe2api, so I guess they'll need to supply additional endpoints with the same JSON payloads. This is not urgent or even required, but I created the issue so it wouldn't get lost when closing the other meta issue.

**Comment by goanpeca** [comment]
>We currently feed it from the npe2api, so I guess they'll need to supply additional endpoints with the same JSON payloads.

Indeed, sine that is a static site, I don't see this happening soonish :-|

**Comment by psobolewskiPhD** [comment]
Could be about dependencies? line -c nvidia or something?
(bad example, i know they've moved into conda-forge)

**Comment by jaimergp** [comment]
I feel that's only for very specific packages (e.g. `pytorch` on Windows, etc), and not a widespread need. As in, this is likely going to cause more issues that it solves. I would be ok with a very specific setting like "Obtain PyTorch from `-c pytorch` on Windows". That said, I think this particular example will be soon gone thanks to https://github.com/conda-forge/pytorch-cpu-feedstock/pull/231

**Comment by hiroalchem** [comment]
My original post was mainly about PyTorch, particularly how to make it easier for users to install it when they want to use a GPU.
